src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -387,11 +414,14 @@ public class DraggedNode
             /// be moved. However, the movement of ILocatable will be limited by 
             /// region and that it cannot be moved beyond the region.</param>
             /// 
-            public DraggedNode(ILocatable locatable, Point2D mouseCursor)
+            public DraggedNode(ILocatable locatable, Point2D mouseCursor, Guid guid)
             {
                 this.locatable = locatable;
                 deltaX = mouseCursor.X - locatable.X;
                 deltaY = mouseCursor.Y - locatable.Y;
+                initialPostiionX = locatable.X;
+                initialPostiionY = locatable.Y;
+                this.guid = guid;
```

Comment:
Would you explain what the guid is set for?

Reply:
If `locatable` can be cast to ModelBase here, maybe guid can be retrived already without adding as a param?

Reply:
This `guid `is to filter the models by its guid when we get the selected nodes/annotations. Like this:
```
ModelBase model = DynamoSelection.Instance.Selection.
                        Where((x) => (x is ModelBase)).Cast<ModelBase>().FirstOrDefault(x => x.GUID == draggedNode.guid);

                    changedPositionModels.Add(model);
```
Now I'm casting it in the constructor, thanks!